streaming/java/streaming-state/src/main/java/org/ray/streaming/state/keystate/desc/AbstractStateDescriptor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ray.streaming.state.keystate.desc;
+
+import com.google.common.base.Preconditions;
+import org.ray.streaming.state.keystate.state.State;
+
+/**
+ * This class defines basic data structures of StateDescriptor.
+ */
+public abstract class AbstractStateDescriptor<S extends State, T> {
+
+  private final String name;
+  private String tableName;
+  private Class<T> type;
+
+  protected AbstractStateDescriptor(String name, Class<T> type) {
+    this.name = name;
+    this.type = type;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public Class<T> getType() {
+    return type;
+  }
+
+  protected Class<T> setType(Class<T> type) {
+    return type;
+  }
+
+  public String getTableName() {
+    return tableName;
+  }
+
+  public void setTableName(String tableName) {
+    this.tableName = tableName;
+  }
+
+  public abstract DescType getDescType();
+
+  public String getIdentify() {
+    Preconditions.checkArgument(this.tableName != null, "table name must not be null.");
+    Preconditions.checkArgument(this.name != null, "table name must not be null.");
+    return this.name;
+  }
+
+  @Override
+  public String toString() {
+    return "AbstractStateDescriptor{" + "tableName='" + tableName + '\'' + ", name='" + name + '\''
+        + ", type=" + type + '}';
+  }
+
+  public enum DescType {
```

Comment:
I see `Desc` and `Descriptor` are mixed. Do we need to using `Descriptor` in all place or using abbr in all place? Otherwise developer will be confused why not this place use full name and vise versa